plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableSearchQuery.php

```diff
@@ -235,6 +224,74 @@ private function generate_where_for_search_filter( string $search_filter ) : str
 		);
 	}
 
+	/**
+	 * Helper function to generate the WHERE clause for products search. Uses FTS when available.
+	 *
+	 * @return string|null WHERE clause for products search.
+	 */
+	private function get_where_for_products() {
+		global $wpdb;
+		$items_table  = $this->query->get_table_name( 'items' );
+		$orders_table = $this->query->get_table_name( 'orders' );
+		$fts_enabled  = get_option( CustomOrdersTableController::HPOS_FTS_INDEX_OPTION ) === 'yes' && get_option( CustomOrdersTableController::HPOS_FTS_ORDER_ITEM_INDEX_CREATED_OPTION ) === 'yes';
+
+		if ( $fts_enabled ) {
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- $orders_table and $items_table are hardcoded.
+			return $wpdb->prepare(
+				"
+$orders_table.id in (
+	SELECT order_id FROM $items_table search_query_items WHERE
+	MATCH ( search_query_items.order_item_name ) AGAINST ( %s IN BOOLEAN MODE )
+)
+",
+				'*' . $wpdb->esc_like( $this->search_term ) . '*'
+			);
+			// phpcs:enable
+		}
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- $orders_table and $items_table are hardcoded.
+		return $wpdb->prepare(
+			"
+$orders_table.id in (
+	SELECT order_id FROM $items_table search_query_items WHERE
+	search_query_items.order_item_name LIKE %s
+)
+",
+			'%' . $wpdb->esc_like( $this->search_term ) . '%'
+		);
+		// phpcs:enable
+	}
+
+	/**
+	 * Helper function to generate the WHERE clause for customers search. Uses FTS when available.
+	 *
+	 * @return string|null WHERE clause for customers search.
+	 */
+	private function get_where_for_customers() {
+		global $wpdb;
+		$order_table   = $this->query->get_table_name( 'orders' );
+		$address_table = $this->query->get_table_name( 'addresses' );
+
+		$fts_enabled = get_option( CustomOrdersTableController::HPOS_FTS_INDEX_OPTION ) === 'yes' && get_option( CustomOrdersTableController::HPOS_FTS_ADDRESS_INDEX_CREATED_OPTION ) === 'yes';
+
+		if ( $fts_enabled ) {
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- $order_table and $address_table are hardcoded.
+			return $wpdb->prepare(
+				"
+$order_table.id IN (
+	SELECT order_id FROM $address_table WHERE
+	MATCH( $address_table.first_name, $address_table.last_name, $address_table.address_1, $address_table.address_2, $address_table.city, $address_table.postcode, $address_table.country, $address_table.state ) AGAINST ( %s IN BOOLEAN MODE )
```

Comment:
This doesn't seem to work correctly. When we create the full-text index on this table, we do it for the following columns: `(first_name, last_name, company, address_1, address_2, city, state, postcode, country, email)` and I believe that's the same set of columns we should be querying here. Otherwise the query fails with `Can't find FULLTEXT index matching the column list`.

Regardless of the error, we definitely want to query e-mail and company too, right?

Reply:
Yes, I think I was testing on a manually created index. Thanks for catching, addressed in https://github.com/woocommerce/woocommerce/pull/46130/commits/31d4b2f06490723ffaeaf6fd5879995db0641756 plus added unit tests.